### PR TITLE
infrastructure change to consistently run functional test with IoC

### PR DESCRIPTION
**Pull request by klawani**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-7278

### Change description

See JIRA Ticket

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -29,11 +29,12 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
 // THESE WILL BE OVERWRITTEN IN ASE DEPLOYMENTS WITH TERRAFORM OUTPUTS.TF
 // THESE CAN BE PICKED UP IN INTEGRATION-TEST CODE DIRECTLY
 env.IDAM_URL = 'https://idam-api.aat.platform.hmcts.net'
-env.CLAIM_STORE_URL = 'http://cmc-claim-store-aat.service.core-compute-aat.internal'
+env.CLAIM_STORE_URL = 'http://cmc-claim-store-aat-staging.service.core-compute-aat.internal'
```
Comment (dharmendrak) [inline Jenkinsfile_CNP]: not sure of this, this was originally for AKS based functional tests, this change is not helping anything

File: docker-compose.functional-tests.yml [hunk 2]
```diff
@@ -6,20 +6,21 @@ services:
       command: test:integration --grep @citizen
       environment:
         - WEB_DRIVER_HOST=remote-webdriver
-        - CITIZEN_APP_URL=${TEST_URL}
```
Comment (dharmendrak) [inline docker-compose.functional-tests.yml]: TEST_URL is used locally for nightly build and security, please search and change in all occurrences .

Comment (klawani) [reply docker-compose.functional-tests.yml]: I will look at that in a seperate PR. However, TEST_URL still available in the Jenkins Library to be used.

File: Jenkinsfile_CNP (hunk 1 above)
Comment (klawani) [reply Jenkinsfile_CNP]: That is correct. I will try to remove it after a first successful build to be honest. Thanks

**Pull request merged**